### ISSUE TITLE
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry when enabling MSI modules

### ISSUE THREAD

**Issue by nemke82** [magento/inventory#2531]
### Preconditions (*)
Freshly upgraded 2.3.2 Magento (from 2.2.9) reports SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry when enabling MSI modules.

### Steps to reproduce (*)
Followed https://devdocs.magento.com/extensions/inventory-management/#previous-magento-versions guidelines to enable MSI modules. When php bin/magento setup:upgrade pushed its results with:

Module 'Magento_InventoryLowQuantityNotification':
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'default-990201' for key 'PRIMARY', query was: INSERT INTO `inventory_low_stock_notification_configuration` SELECT `source_item`.`source_code`, `source_item`.`sku`, `stock_item`.`notify_stock_qty` FROM `cataloginventory_stock_item` AS `stock_item`
 INNER JOIN `catalog_product_entity` AS `product` ON product.entity_id = stock_item.product_id
 INNER JOIN `inventory_source_item` AS `source_item` ON source_item.sku = product.sku WHERE (stock_item.use_config_notify_stock_qty = 0) AND (source_item.source_code = 'default')

To test which SKU involved:
mysql> INSERT INTO `inventory_low_stock_notification_configuration` SELECT `source_item`.`source_code`, `source_item`.`sku`, `stock_item`.`notify_stock_qty` FROM `cataloginventory_stock_item` AS `stock_item`  INNER JOIN `catalog_product_entity` AS `product` ON product.entity_id = stock_item.product_id  INNER JOIN `inventory_source_item` AS `source_item` ON source_item.sku = product.sku WHERE (stock_item.use_config_notify_stock_qty = 0) AND (source_item.source_code = 'default');
ERROR 1062 (23000): Duplicate entry 'default-112301' for key 'PRIMARY'
mysql> select * from inventory_source_item where sku = 112301;
+----------------+-------------+--------+----------+--------+
| source_item_id | source_code | sku    | quantity | status |
+----------------+-------------+--------+----------+--------+
|            239 | default     | 112301 |   0.0000 |      0 |
+----------------+-------------+--------+----------+--------+
1 row in set, 5 warnings (0.00 sec)

### Expected result (*)
php bin/magento setup:upgrade to pass and move on with my life...

### Actual result (*)
Module 'Magento_InventoryLowQuantityNotification':
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'default-990201' for key 'PRIMARY', query was: INSERT INTO `inventory_low_stock_notification_configuration` SELECT `source_item`.`source_code`, `source_item`.`sku`, `stock_item`.`notify_stock_qty` FROM `cataloginventory_stock_item` AS `stock_item`
 INNER JOIN `catalog_product_entity` AS `product` ON product.entity_id = stock_item.product_id
 INNER JOIN `inventory_source_item` AS `source_item` ON source_item.sku = product.sku WHERE (stock_item.use_config_notify_stock_qty = 0) AND (source_item.source_code = 'default')

**FIX WAS TO REMOVE THAT SKU from inventory_source_item table.**

Output:
mysql> delete from inventory_source_item where sku = 112301;
Query OK, 1 row affected, 5 warnings (0.00 sec)

Don't like my fix so wondering what else can go wrong here or if this can be somehow corrected in next versions to prevent similar happening. Cheers.

**Comment by smoskaluk** [comment]
Hi @nemke82 , thank you for your report, but I couldn't reproduce this issue.  Do you have Magento Community or Enterprise edition?

**Comment by nemke82** [comment]
Hello @smoskaluk. It is Magento Community 2.3.2 version freshly upgraded from 2.2.9 as I've said in my previous reply. When doing Setup:upgrade it fails with Module 'Magento_InventoryLowQuantityNotification'. Now probably you can't replicate on the fresh 2.3.2 as this requires 2.2.9 with sample data then upgraded to 2.3.2 and inventory modules enabled as per  https://devdocs.magento.com/extensions/inventory-management/#previous-magento-versions guidelines.

**Comment by smoskaluk** [comment]
Hi @nemke82! I did upgrade from Magento 2.2.9 with sample data  to 2.3.2 and then enabled inventory modules - everything run well, without any failure

**Comment by ishakhsuvarov** [comment]
Closing for now due to inactivity

**Comment by HansiBlomberg** [comment]
I experienced the same exact issue when enabling the inventory modules on Magento Community Edition 2.3.5-p1.  My solution was to disable one module in app/etc/config.php
'Magento_InventoryLowQuantityNotification' => 1, 
change to
'Magento_InventoryLowQuantityNotification' => 0

Then I went to backend and removed the product causing the duplicate item error.
In my case it was a configurable product, I removed both the "main" product and the "child" products.

Then I enabled the module and ran setup:upgrade again with no errors.
